experimental/ecma402_listformat/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+    ///
+    ///   [bcp47]: https://tools.ietf.org/html/bcp47
+    fn to_bcp47() -> &'a str;
+}
+
+/// The [listformat] mod contains all the needed implementation bits for `Intl.ListFormat`.
+///
+/// > Note: This is not yet the entire API.  I'd like to get to a consensus on what has been
+/// defined, then use the patterns adopted here for the rest.
+pub mod listformat {
+    /// Contains the API configuration as prescribed by ECMA 402.
+    ///
+    /// The meaning of the options is the same as in the similarly named
+    /// options in the JS version.
+    ///
+    /// See [options::In] for the contents of the options.  See the [Factory]
+    /// trait for the use of the options.
+    pub mod options {
+        /// Chooses the list formatting approach.
+        #[derive(Eq, PartialEq, Debug, Clone)]
+        pub enum Style {
+            Long,
+            Short,
+            Narrow,
+        }
+        /// Chooses between "this, that and other", and "this, that or other".
+        #[derive(Eq, PartialEq, Debug, Clone)]
+        pub enum Type {
+            /// "This, that and other".
+            Conjunction,
+            /// "This, that or other".
+            Disjunction,
+        }
+
+        /// The options set by the user at construction time.  See discussion at the top level
+        /// about the name choice.  Provides as a "bag of options" since we don't expect any
+        /// implementations to be attached to this struct.
+        #[derive(Eq, PartialEq, Debug, Clone)]
+        pub struct In {
+            /// Selects a [Style] for the formatted list.
+            pub style: Style,
+            /// Selects a [Type] for the formatted list.
+            pub in_type: Type,
+        }
+    }
+
+    /// A factory object containing the methods used to construct new [Format], as well
+    /// as associated functions prescribed by `Intl.ListFormat`.
+    ///
+    /// While ECMA 402 originally has functions under `Intl`, we probably want to
+    /// obtain a separate factory from each implementor.
+    pub trait Factory<'a> {
+        /// A better name for this associated type is most welcome.
+        type Ret: Format<'a>;
+
+        /// The type of error reported, if any.
+        type Error: std::error::Error;
+
+        /// Creates a new [Format].
+        ///
+        /// Creation may fail, for example, if the locale-specific data is not loaded, or if
+        /// the supplied options are inconsistent.
+        fn try_new(l: impl crate::Locale<'a>, opts: options::In) -> Result<Self::Ret, Self::Error>;
+
+        /// The type used to implement [crate::Locale].
+        type Locale: crate::Locale<'a> + 'a;
+
+        /// The type of iterator returned by the call to `supported_locales_of`.
+        type LocaleIter: std::iter::Iterator<Item = &'a Self::Locale>;
+
+        fn supported_locales_of<I>(l: I) -> Result<Self::LocaleIter, Self::Error>
+        where
+            I: IntoIterator<Item = &'a Self::Locale>;
```

Comment:
Same here, why not `IntoIterator<Item = Locale>` ?

Reply:
Done.